diff --git a/dnf-plugin-spacewalk.spec b/dnf-plugin-spacewalk.spec
index 17df012..e20e59f 100644
--- a/dnf-plugin-spacewalk.spec
+++ b/dnf-plugin-spacewalk.spec
@@ -12,7 +12,7 @@
 Summary: DNF plugin for Spacewalk
 Name: dnf-plugin-spacewalk
 Version: 2.8.5
-Release: 6%{?dist}
+Release: 7%{?dist}
 License: GPLv2
 Source0: https://github.com/spacewalkproject/spacewalk/archive/%{name}-%{version}.tar.gz
 URL:     https://github.com/spacewalkproject/spacewalk
@@ -50,6 +50,7 @@ Summary: DNF plugin for Spacewalk
 BuildRequires: python3-devel
 Requires: %{name} = %{version}-%{release}
 Requires: python3-rhn-client-tools >= 2.8.4
+Requires: python3-librepo
 
 %description -n python3-%{name}
 Python 3 specific files for %{name}.
@@ -110,6 +111,9 @@ install -m 644 actions/errata.py %{buildroot}%{python3_sitelib}/rhn/actions/
 %endif
 
 %changelog
+* Wed Oct 31 2018 Tomas Kasparek <tkasparek@redhat.com> 2.8.5-7
+- Resolves: rhbz#1633298 - librepo dependency is not added automatically (michael.mraka@redhat.com)
+
 * Tue Oct 16 2018 Tomas Kasparek <tkasparek@redhat.com> 2.8.5-6
 - spec cleanup (no more builds on Fedora <= 25) (michael.mraka@redhat.com)
 - Resolves: #1637980 - fixed plugin for dnf > 3.6.0 (michael.mraka@redhat.com)
